girder-dandi-archive/tests/rest/test_get_dandiset_stats.py

```diff
@@ -0,0 +1,138 @@
+    resp = server.request(
+        "/folder",
+        method="POST",
+        params={"parentType": "folder", "parentId": dandi_id, "name": "Test Folder"},
+        **request_auth,
+    )
+    assertStatusOk(resp)
+    resp = server.request(f"/dandi/{identifier}/stats", method="GET")
+    assertStatusOk(resp)
+    assert {"bytes": 0, "folders": 1, "items": 0} == resp.json
+
+
+def test_get_dandiset_stats_one_item(server, dandiset_1, request_auth):
+    dandi_id = dandiset_1["_id"]
+    identifier = dandiset_1["name"]
+    resp = server.request(
+        "/item", method="POST", params={"folderId": dandi_id, "name": "Test Item"}, **request_auth,
+    )
+    assertStatusOk(resp)
+    resp = server.request(f"/dandi/{identifier}/stats", method="GET")
+    assertStatusOk(resp)
+    assert {"bytes": 0, "folders": 0, "items": 1} == resp.json
+
+
+def test_get_dandiset_stats_one_file(server, dandiset_1, request_auth, fsAssetstore):  # noqa: N803
+    dandi_id = dandiset_1["_id"]
+    identifier = dandiset_1["name"]
+    resp = server.request(
+        "/item", method="POST", params={"folderId": dandi_id, "name": "Test Item"}, **request_auth,
+    )
+    assertStatusOk(resp)
+    item_id = resp.json["_id"]
+    content = "This is test text."
+    resp = server.request(
+        "/file",
+        method="POST",
+        params={
+            "parentType": "item",
+            "parentId": item_id,
+            "name": "foo.txt",
+            "mimeType": "text/plain",
+            "size": len(content),
+        },
+        body=content,
+        type="text/plain",
+        isJson=False,
+        **request_auth,
+    )
+    assertStatusOk(resp)
+    resp = server.request(f"/dandi/{identifier}/stats", method="GET")
+    assertStatusOk(resp)
+    assert {"bytes": len(content), "folders": 0, "items": 1} == resp.json
+
+
+def test_get_dandiset_stats_tree(server, dandiset_1, request_auth, fsAssetstore):  # noqa: N803
+    def mkfolder(name, parent_id):
+        resp = server.request(
+            "/folder",
+            method="POST",
+            params={"parentType": "folder", "parentId": parent_id, "name": name},
+            **request_auth,
+        )
+        assertStatusOk(resp)
+        return resp.json["_id"]
+
+    def mkitem(name, folder_id):
+        resp = server.request(
+            "/item", method="POST", params={"folderId": folder_id, "name": name}, **request_auth,
+        )
+        assertStatusOk(resp)
+        return resp.json["_id"]
+
+    def mkfile(name, item_id, content):
+        resp = server.request(
+            "/file",
+            method="POST",
+            params={
+                "parentType": "item",
+                "parentId": item_id,
+                "name": name,
+                "mimeType": "text/plain",
+                "size": len(content),
+            },
+            body=content,
+            type="text/plain",
+            isJson=False,
+            **request_auth,
+        )
+        assertStatusOk(resp)
```

Comment:
These helpers would be useful in the other test methods. Can you move these to the top of the file and invoke them in the other test methods?